### PR TITLE
Add lint for statics with explicit static lifetime.

### LINKED ISSUE #4138: Check for static items with 'static lifetime
It just like https://rust-lang.github.io/rust-clippy/master/index.html#const_static_lifetime but for `static` item.

### Example

```rust
pub static FOO: &'static [u8] = b"bar";
```

Could be rewritten as:

```rust
pub static FOO: &[u8] = b"bar";
```

---

`cargo clippy -V` output: clippy 0.0.212 (60a609a 2019-05-17)

### PR DESCRIPTION
**Pull request by krk**
changelog: Add lint for statics with explicit static lifetime, fixes #4138.

### CODE REVIEW

**Comment by flip1995** [comment]
Please run `util/dev update_lints`

**Comment by oli-obk** [comment]
does anything speak against just extending the lint for constants to cover statics, too? https://rust-lang.github.io/rust-clippy/master/index.html#const_static_lifetime

**Comment by krk** [comment]
@flip1995 Running `util/dev update_lints` did not introduce any changes.

**Comment by flip1995** [comment]
Travis disagrees: https://travis-ci.com/rust-lang/rust-clippy/jobs/204695122#L1283

```
Not all lints defined properly. Please run `util/dev update_lints` to make sure all lints are defined properly.
```

CHANGELOG and README need updates.

But I agree with https://github.com/rust-lang/rust-clippy/pull/4162#issuecomment-498222446. At least you should be able reuse the logic there?

**Comment by krk** [comment]
I have rebased and pushed, travis is green now, will try to reuse the logic.

**Comment by flip1995** [comment]
Thanks! Can you put the code of the lints into the same file (and into the same `EarlyLintPass`)? You can rename the file, if you want (If you do, please rename in a separate commit).

**Comment by flip1995** [comment]
You should also be able to use the Visitor provided by `rustc`: https://doc.rust-lang.org/nightly/nightly-rustc/syntax/visit/trait.Visitor.html?search=#method.visit_ty, instead of rewriting portions of it.

**Comment by krk** [comment]
> You should also be able to use the Visitor provided by `rustc`: https://doc.rust-lang.org/nightly/nightly-rustc/syntax/visit/trait.Visitor.html?search=#method.visit_ty, instead of rewriting portions of it.

`visit_type` method in `clippy_lints/src/const_static_lifetime.rs` existed before this PR and I am reluctant to include changing it to the `rustc` visitor here.

File: clippy_lints/src/redundant_static_lifetime.rs [hunk 1]
```diff
@@ -16,29 +16,32 @@ declare_clippy_lint! {
...
 }
 
-declare_lint_pass!(StaticConst => [CONST_STATIC_LIFETIME]);
+declare_lint_pass!(RedundantStaticLifetime => [REDUNDANT_STATIC_LIFETIME]);
 
-impl StaticConst {
+impl RedundantStaticLifetime {
     // Recursively visit types
-    fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>) {
+    pub fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>, reason: &str) {
```
Comment (flip1995) [inline clippy_lints/src/redundant_static_lifetime.rs]:
```suggestion
    fn visit_type(&mut self, ty: &Ty, cx: &EarlyContext<'_>, reason: &str) {
```

This `pub` shouldn't be necessary.

**Comment by flip1995** [comment]
That's ok, you can use the existing function! 

LGTM, just one NIT and this can get merged.

**Comment by flip1995** [comment]
Thanks!

@bors r+

**Pull request merged**